engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java

```diff
@@ -460,7 +460,26 @@ protected void restoreOutput(PrintStream originalStream)
     }
   }
 
-  AppPackage newAppPackageInstance(File f) throws IOException, ZipException
+  AppPackage newAppPackageInstance(URI uri, boolean suppressOutput) throws IOException
+  {
+    PrintStream outputStream = suppressOutput ? suppressOutput() : null;
+    try {
+      final String scheme = uri.getScheme();
+      if (scheme == null || scheme.equals("file")) {
+        return new AppPackage(new FileInputStream(new File(expandFileName(uri.getPath(), true))), true);
+      } else {
```

Comment:
It is not handled in other places as well. The goal of the PR is to add support for files on hdfs only. Support for other schemes may be done as a separate PR.